File: audiobook/src/main/java/de/ph1b/audiobook/fragment/BaseFragment.java
package de.ph1b.audiobook.fragment;

import android.support.v4.app.Fragment;

import de.ph1b.audiobook.utils.App;

/**
 * Base fragment all fragments should inherit from. Handles memory leak detection.
 *
 * @author Paul Woitaschek
 */
public class BaseFragment extends Fragment {

    @Override
    public void onDestroy() {
        super.onDestroy();

        App.leakWatch(this);
    }
}

File: audiobook/src/main/kotlin/de/ph1b/audiobook/fragment/BaseFragment.kt
package de.ph1b.audiobook.fragment

import android.support.v4.app.Fragment
import de.ph1b.audiobook.injection.App

/**
 * Base fragment all fragments should inherit from. Handles memory leak detection.
 *
 * @author Paul Woitaschek
 */
open class BaseFragment : Fragment() {

    override fun onDestroy() {
        super.onDestroy()

        App.leakWatch(this)
    }
}